xicam/SAXS/processing/correction.py

```diff
@@ -1,56 +1,118 @@
-from csxtools.fastccd import correct_images
-from csxtools.utils import stackmean
+from functools import partial
+
 import numpy as np
-from xicam.plugins.processingplugin import Input, Output, ProcessingPlugin
-
-
-class CSXCorrectImage(ProcessingPlugin):
-    """Subtract backgrond and gain correct images
-    This routine subtrtacts the backgrond and corrects the images
-    for the multigain FastCCD ADC.
-    Parameters
-    ----------
-    in : array_like
-        Input array of images to correct of shape (N, y, x)  where N is the
-        number of images and x and y are the image size.
-    dark : array_like, optional
-        Input array of dark images. This should be of shape (3, y, x).
-        dark[0] is the gain 8 (most sensitive setting) dark image with
-        dark[2] being the gain 1 (least sensitive) dark image.
-    flat : array_like, optional
-        Input array for the flatfield correction. This should be of shape
-        (y, x)
-    gain : tuple, optional
-        These are the gain multiplication factors for the three different
-        gain settings
-    Returns
-    -------
-    array_like
-        Array of corrected images of shape (N, y, x)
-"""
-    # Images are expected to be of type np.uint16 (csxtools.utils.py:94)
-    bitmasked_images = Input(description="", type=np.ndarray, visible=False)
-    # Darks are expected to be of type np.float32 (csxtools.fastccd.images.py:43)
-    dark_images = Input(description="", type=np.ndarray, visible=False)
-    # Flats are expected to be of type np.float32 (csxtools.fastccd.images.py:47)
-    flat_field = Input(description="", type=np.ndarray, default=None, visible=False)
-    gain = Input(description="", type=tuple, visible=False)
-
-    corrected_images = Output(description="", type=np.ndarray)
+
+from xicam.core import msg
+from xicam.plugins import ProcessingPlugin, Input, Output
+
+
+class CorrectImage(ProcessingPlugin):
```

Comment:
Rename to CorrectFastCCDImage (correction is unique to fastccd)

Reply:
good catch! refactored